In [3]:
%matplotlib inline
from pycbc import filter
import matplotlib.pyplot as plt
from pycbc.waveform import get_td_waveform
from pycbc.detector import Detector
import pylab
from pycbc.waveform import td_approximants, fd_approximants

import os
import h5py
import numpy as np
import pandas as pd
import seaborn as sns


from gwpy.time import to_gps, from_gps
from gwpy.table import Table
from gwpy.time import tconvert
from gwdetchar import scattering
from gwpy.table import EventTable
from gwpy.segments import Segment
from gwpy.segments import SegmentList
from gwpy.timeseries import TimeSeries
from gwtrigfind import find_trigger_files
from gwpy.segments import DataQualityDict
from gwpy.segments import DataQualityFlag
from gwpy.timeseries import TimeSeriesDict
from gwpy.spectrogram import Spectrogram

import glob
import time
import datetime
from datetime import timedelta
from itertools import islice
from scipy.signal import correlate
import matplotlib.pyplot as plt
from matplotlib.ticker import FormatStrFormatter

In [2]:
def gen_thresholds(dataframe, thresholds = [2,1,1]):
    
    df = dataframe
    if len(thresholds) < 4:
        thresh_cond = (df['t1'] > thresholds[0]) | (df['t2'] > thresholds[1]) | (df['t3'] > thresholds[2]) 
    else:
        thresh_cond = (df['t1']>thresholds[0]) | (df['t2']>thresholds[1]) |(df['t3']>thresholds[2]) | (df['t4']>thresholds[3]) | (df['t5'] > thresholds[4])
        
    return thresh_cond

### Injections and background, overlap and non overlap analysis

In [4]:
df_inj_ovlp = pd.read_csv('files/combined_bbh_new.csv')
df_inj_nonovlp = pd.read_csv('files/combined_bbh_mutexc.csv')

In [5]:
df_bkg_ovlp = pd.read_csv('files/combined_bkg_ovlp.csv')
df_bkg_nonovlp = pd.read_csv('files/combined_bkg_nonovlp.csv')

In [8]:
df_bkg_nonovlp['filename'].nunique()

1591

#### The files df_inj_ovlp, df_inj_nonovlp, df_bkg_ovlp and df_bkg_nonovlp contain overlap and non overlap analysis with energy threshold of 60 for the 1354 injections and 1591 background events
The injection files on which the calc_pixel_occupancy was ran to produce df_inj_ovlp and df_inj_nonovlp are stored at '/home/olib/public_html/Ethan/O3_replay_inj/0lag_lowfreq/q_transforms/BBH_data_snr_5/' and the background files on which the calc_pixel_occupancy was ran to produce df_bkg_ovlp and df_bkg_nonovlp is stored at '/home/ethan.marx/forks/retraction_scripts/projects/data/background/raw/'

### Attributes of injections and background

In [6]:
df_inj_attr = pd.read_csv('files/combined_bbh_new_attrs.csv')

In [7]:
df_bkg_attr = pd.read_csv('files/combined_bkg_attrs_nonovlp.csv')

### The dataframes df_inj_attr and df_bkg_attr contain attributes of the 1354 injections and 1591 background events. 

In [9]:
df_inj_attr

replay_time        time_H  coa_phase   eff_dist_H  \
0     1.324070e+09  1.263782e+09   4.400568   658.213588   
1     1.322883e+09  1.262595e+09   5.545442   512.341438   
2     1.323322e+09  1.263034e+09   2.085905  1146.963429   
3     1.325999e+09  1.265711e+09   2.555068   439.029648   
4     1.324281e+09  1.263993e+09   6.249260   861.692729   
...            ...           ...        ...          ...   
1349  1.324834e+09  1.264546e+09   3.709780   542.083678   
1350  1.323042e+09  1.262754e+09   2.631282  1004.274519   
1351  1.324072e+09  1.263784e+09   3.237331   889.423392   
1352  1.323139e+09  1.262851e+09   0.769810   849.112932   
1353  1.325830e+09  1.265542e+09   3.778717   832.866510   

                    approximant     chi_p         z     distance    spin2z  \
0     b'IMRPhenomPv2_NRTidalv2'  0.489192  0.205563  1048.320953 -0.083213   
1     b'IMRPhenomPv2_NRTidalv2'  0.552983  0.159667   792.627414 -0.099114   
2     b'IMRPhenomPv2_NRTidalv2'  0.074477  0.323853  1757.498679 -0.011599   
3     b'IMRPhenomPv2_NRTidalv2'  0.222893  0.201676  1026.221976  0.013461   
4     b'IMRPhenomPv2_NRTidalv2'  0.285242  0.202187  1029.121154 -0.712433   
...                         ...       ...       ...          ...       ...   
1349  b'IMRPhenomPv2_NRTidalv2'  0.345502  0.295165  1579.237687 -0.216031   
1350  b'IMRPhenomPv2_NRTidalv2'  0.573371  0.372609  2069.033503 -0.083389   
1351  b'IMRPhenomPv2_NRTidalv2'  0.531267  0.324816  1763.546735 -0.187699   
1352  b'IMRPhenomPv2_NRTidalv2'  0.721460  0.325681  1768.988754 -0.360885   
1353  b'IMRPhenomPv2_NRTidalv2'  0.443020  0.369869  2051.247807 -0.429045   

      fref  ...         Mc  lambda1       mass2  flow  \
0     10.0  ...  12.682835      0.0   13.432107  10.0   
1     10.0  ...  13.765229      0.0   13.922637  10.0   
2     10.0  ...  27.222245      0.0   30.848315  10.0   
3     10.0  ...  91.721406      0.0  102.310778  10.0   
4     10.0  ...  35.023265      0.0   29.641763  10.0   
...    ...  ...        ...      ...         ...   ...   
1349  10.0  ...  38.210390      0.0   36.210617  10.0   
1350  10.0  ...  87.795678      0.0   84.562523  10.0   
1351  10.0  ...  59.498874      0.0   50.079667  10.0   
1352  10.0  ...  73.213026      0.0   73.168037  10.0   
1353  10.0  ...  49.139647      0.0   55.291628  10.0   

      logpdraw_right_ascension_declination  \
0                                -2.824640   
1                                -3.313658   
2                                -3.504506   
3                                -2.591834   
4                                -2.930356   
...                                    ...   
1349                             -2.675384   
1350                             -3.033623   
1351                             -2.660177   
1352                             -2.834519   
1353                             -2.887629   

      logpdraw_z_mass1_source_mass2_source_spin1x_spin1y_spin1z_spin2x_spin2y_spin2z  \
0                                            -11.318613                                
1                                            -12.801971                                
2                                             -2.906306                                
3                                             -9.378661                                
4                                            -12.596220                                
...                                                 ...                                
1349                                         -11.734877                                
1350                                         -14.318081                                
1351                                         -14.794976                                
1352                                         -15.555550                                
1353                                         -12.418400                                

      logpdraw_lambda1_lambda2_GIVEN_mass1_source_mass

In [10]:
df_bkg_attr

ifo_freq  fmin  fres        ifar          m1         m2          time  \
0            0  10.0  0.05   70.549339   18.581810  13.023375  1.248149e+09   
1            0  10.0  0.05   79.977180    9.023204   5.538040  1.247408e+09   
2            0  10.0  0.05   77.122160   15.986259   7.134080  1.252482e+09   
3            0  10.0  0.05  416.761820   20.475435   7.730863  1.247266e+09   
4            0  10.0  0.05   68.722388  489.172450   9.301118  1.238307e+09   
...        ...   ...   ...         ...         ...        ...           ...   
1586         0  10.0  0.05   79.632953   46.586437   9.287184  1.238176e+09   
1587         0  10.0  0.05  165.747140   66.721344   5.479237  1.246106e+09   
1588         0  10.0  0.05   83.958693    7.565491   5.176327  1.241144e+09   
1589         0  10.0  0.05   87.723144  492.076930   5.817298  1.252567e+09   
1590         0  10.0  0.05  443.930831  466.255830  10.088473  1.238406e+09   

      tres  window                              filename  
0     0.01     2.0  acb180ce-4845-4652-9139-6752941fc844  
1     0.01     2.0  3c99d9bc-5103-4c52-8a49-c31e4114e8a5  
2     0.01     2.0  8ba4bd72-8d07-4986-925d-229e20431c04  
3     0.01     2.0  ee9849cf-91a5-45e6-adef-e1ffa556a72f  
4     0.01     2.0  16df310e-9972-4b2a-be1a-a67338b3accd  
...    ...     ...                                   ...  
1586  0.01     2.0  c579ddb2-3487-4f5a-aad0-76496e2d7be0  
1587  0.01     2.0  0f6e6114-0da9-41ca-a1cb-fbdd7757efe5  
1588  0.01     2.0  b5719a8b-4143-4ef7-9c03-8dcd316c56c5  
1589  0.01     2.0  7f5e61ea-cd1e-4572-ac73-cfae0ea2519d  
1590  0.01     2.0  d8880d5a-9b18-4a3a-816c-3a12341f4ec7  

[1591 rows x 10 columns]

### Let's now look at injections flagged first by non overlap method and then by overlap method

In [11]:
df_inj_nonovlp_flag_a = df_inj_nonovlp[gen_thresholds(df_inj_nonovlp, thresholds=[0,0,4,0,0])]

In [12]:
df_inj_nonovlp_flag_a.head()

ifo_freq   t1   t2   t3   t4   t5        time
12     L1_f1  0.0  0.0  0.3  0.1  5.0  1325566299
13     L1_f2  0.0  0.0  0.0  0.2  0.0  1325566299
30     L1_f1  0.0  0.4  1.2  0.0  0.5  1325410502
408    L1_f1  2.1  0.0  0.0  1.2  2.3  1322690262
411    H1_f1  0.0  1.0  0.3  0.0  2.1  1322690262

df_inj_nonovlp_flag_a are events flagged by non overlap method with the thresholds [0,0,4,0,0]

In [13]:
df_inj_nonovlp_flag_a_inovlp = df_inj_ovlp[df_inj_ovlp['time'].isin(df_inj_nonovlp_flag_a['time'])]

In [14]:
df_inj_nonovlp_flag_a_inovlp

ifo_freq   t1   t2   t3        time
198     L1_f1  5.4  2.7  1.3  1324331465
199     L1_f2  0.7  0.3  0.2  1324331465
200     L1_f3  0.0  0.0  0.0  1324331465
201     H1_f1  1.8  0.9  0.5  1324331465
202     H1_f2  0.5  0.2  0.1  1324331465
...       ...  ...  ...  ...         ...
8065    L1_f2  0.0  0.0  0.0  1325865224
8066    L1_f3  0.0  0.0  0.0  1325865224
8067    H1_f1  0.0  0.0  0.0  1325865224
8068    H1_f2  0.0  0.0  0.0  1325865224
8069    H1_f3  0.0  0.0  0.0  1325865224

[618 rows x 5 columns]

df_inj_nonovlp_flag_a_inovlp is the part of df_inj_ovlp which is in df_inj_nonovlp_flag_a, we will now apply the overlap threhsolds on this to find events flagged by both

In [15]:
df_inj_both_flag_a = df_inj_nonovlp_flag_a_inovlp[gen_thresholds(df_inj_nonovlp_flag_a_inovlp, thresholds=[4,2,2])]

In [16]:
df_inj_both_flag_a.head()

ifo_freq   t1   t2   t3        time
198     L1_f1  5.4  2.7  1.3  1324331465
306     L1_f1  3.6  2.3  1.1  1325581835
330     L1_f1  5.6  2.8  1.4  1322914217
936     L1_f1  4.2  2.1  1.0  1324788826
1332    L1_f1  8.9  4.7  2.3  1325614061

The information in the dataframe df_inj_both_flag_a is also stored in the csv file /home/siddharth.soni/Retraction_project/Retraction_paper/inj_abvnonvlp_inovlp.csv

df_inj_flagged_both_a are the events that are flagged by both, first by nonoverlap and then by overlap method with thresholds ([0,0,4,0,0] and [4,2,2]) respectively, energy threshold 60

In [37]:
#df_inj_flagged_both_a = pd.read_csv('/home/siddharth.soni/Retraction_project/Retraction_paper/inj_abvnonvlp_inovlp.csv')

### Injection attributes

In [17]:
df_inj_attr

replay_time        time_H  coa_phase   eff_dist_H  \
0     1.324070e+09  1.263782e+09   4.400568   658.213588   
1     1.322883e+09  1.262595e+09   5.545442   512.341438   
2     1.323322e+09  1.263034e+09   2.085905  1146.963429   
3     1.325999e+09  1.265711e+09   2.555068   439.029648   
4     1.324281e+09  1.263993e+09   6.249260   861.692729   
...            ...           ...        ...          ...   
1349  1.324834e+09  1.264546e+09   3.709780   542.083678   
1350  1.323042e+09  1.262754e+09   2.631282  1004.274519   
1351  1.324072e+09  1.263784e+09   3.237331   889.423392   
1352  1.323139e+09  1.262851e+09   0.769810   849.112932   
1353  1.325830e+09  1.265542e+09   3.778717   832.866510   

                    approximant     chi_p         z     distance    spin2z  \
0     b'IMRPhenomPv2_NRTidalv2'  0.489192  0.205563  1048.320953 -0.083213   
1     b'IMRPhenomPv2_NRTidalv2'  0.552983  0.159667   792.627414 -0.099114   
2     b'IMRPhenomPv2_NRTidalv2'  0.074477  0.323853  1757.498679 -0.011599   
3     b'IMRPhenomPv2_NRTidalv2'  0.222893  0.201676  1026.221976  0.013461   
4     b'IMRPhenomPv2_NRTidalv2'  0.285242  0.202187  1029.121154 -0.712433   
...                         ...       ...       ...          ...       ...   
1349  b'IMRPhenomPv2_NRTidalv2'  0.345502  0.295165  1579.237687 -0.216031   
1350  b'IMRPhenomPv2_NRTidalv2'  0.573371  0.372609  2069.033503 -0.083389   
1351  b'IMRPhenomPv2_NRTidalv2'  0.531267  0.324816  1763.546735 -0.187699   
1352  b'IMRPhenomPv2_NRTidalv2'  0.721460  0.325681  1768.988754 -0.360885   
1353  b'IMRPhenomPv2_NRTidalv2'  0.443020  0.369869  2051.247807 -0.429045   

      fref  ...         Mc  lambda1       mass2  flow  \
0     10.0  ...  12.682835      0.0   13.432107  10.0   
1     10.0  ...  13.765229      0.0   13.922637  10.0   
2     10.0  ...  27.222245      0.0   30.848315  10.0   
3     10.0  ...  91.721406      0.0  102.310778  10.0   
4     10.0  ...  35.023265      0.0   29.641763  10.0   
...    ...  ...        ...      ...         ...   ...   
1349  10.0  ...  38.210390      0.0   36.210617  10.0   
1350  10.0  ...  87.795678      0.0   84.562523  10.0   
1351  10.0  ...  59.498874      0.0   50.079667  10.0   
1352  10.0  ...  73.213026      0.0   73.168037  10.0   
1353  10.0  ...  49.139647      0.0   55.291628  10.0   

      logpdraw_right_ascension_declination  \
0                                -2.824640   
1                                -3.313658   
2                                -3.504506   
3                                -2.591834   
4                                -2.930356   
...                                    ...   
1349                             -2.675384   
1350                             -3.033623   
1351                             -2.660177   
1352                             -2.834519   
1353                             -2.887629   

      logpdraw_z_mass1_source_mass2_source_spin1x_spin1y_spin1z_spin2x_spin2y_spin2z  \
0                                            -11.318613                                
1                                            -12.801971                                
2                                             -2.906306                                
3                                             -9.378661                                
4                                            -12.596220                                
...                                                 ...                                
1349                                         -11.734877                                
1350                                         -14.318081                                
1351                                         -14.794976                                
1352                                         -15.555550                                
1353                                         -12.418400                                

      logpdraw_lambda1_lambda2_GIVEN_mass1_source_mass

### Let's now look at the attributes of injections flagged by both non overlap and overlap methods

In [18]:
df_inj_both_flag_a_attrs = df_inj_attr[df_inj_attr['replay_time'].astype(int).isin(df_inj_both_flag_a['time'].unique())]

In [19]:
df_inj_both_flag_a_attrs

replay_time        time_H  coa_phase   eff_dist_H  \
0     1.324070e+09  1.263782e+09   4.400568   658.213588   
50    1.324757e+09  1.264469e+09   6.106272  1376.035529   
111   1.324846e+09  1.264558e+09   1.488794  1544.489371   
168   1.322665e+09  1.262377e+09   5.271821  1374.320669   
213   1.322688e+09  1.262400e+09   0.891948  3031.443021   
224   1.323209e+09  1.262921e+09   1.593250  2095.446538   
253   1.322914e+09  1.262626e+09   5.121788   229.745252   
300   1.324978e+09  1.264690e+09   4.063662   744.723446   
316   1.325582e+09  1.265294e+09   1.050888   486.222231   
317   1.322692e+09  1.262404e+09   5.037607    59.415664   
422   1.322953e+09  1.262665e+09   5.879188   213.549844   
443   1.324331e+09  1.264043e+09   0.637793   263.261272   
461   1.324598e+09  1.264310e+09   1.868832    65.748986   
545   1.323384e+09  1.263096e+09   4.108063   330.514768   
574   1.323530e+09  1.263242e+09   3.220790   104.782218   
611   1.322654e+09  1.262366e+09   4.378828   170.019015   
658   1.324789e+09  1.264501e+09   5.677676   142.878499   
737   1.323159e+09  1.262871e+09   2.437367   423.274846   
805   1.324391e+09  1.264103e+09   3.539436   226.300142   
824   1.325201e+09  1.264913e+09   2.200701   172.219706   
836   1.325142e+09  1.264854e+09   5.414127   592.295541   
957   1.323122e+09  1.262834e+09   4.801792  1588.073777   
1022  1.324605e+09  1.264317e+09   1.809101    87.073637   
1031  1.323050e+09  1.262762e+09   4.027301   915.820318   
1103  1.322665e+09  1.262377e+09   2.698748  1498.913325   
1106  1.324903e+09  1.264615e+09   3.826657   136.629306   
1130  1.322896e+09  1.262608e+09   4.666997   211.073200   
1157  1.325150e+09  1.264862e+09   3.435548   934.515547   
1166  1.322645e+09  1.262357e+09   5.959291   811.615611   
1243  1.325614e+09  1.265326e+09   3.478056    83.184184   
1280  1.322603e+09  1.262315e+09   2.892433   225.453715   
1326  1.325566e+09  1.265278e+09   4.635718  2557.224756   

                    approximant     chi_p         z     distance    spin2z  \
0     b'IMRPhenomPv2_NRTidalv2'  0.489192  0.205563  1048.320953 -0.083213   
50    b'IMRPhenomPv2_NRTidalv2'  0.165124  0.293007  1565.981419  0.268909   
111   b'IMRPhenomPv2_NRTidalv2'  0.708649  0.295344  1580.334641 -0.045178   
168   b'IMRPhenomPv2_NRTidalv2'  0.377101  0.290754  1552.172509 -0.141551   
213   b'IMRPhenomPv2_NRTidalv2'  0.048779  0.646509  3992.662660 -0.008014   
224   b'IMRPhenomPv2_NRTidalv2'  0.274955  0.529894  3140.708789  0.023824   
253   b'IMRPhenomPv2_NRTidalv2'  0.183640  0.078681   370.816808  0.001646   
300   b'IMRPhenomPv2_NRTidalv2'  0.232550  0.264845  1395.086510  0.094368   
316   b'IMRPhenomPv2_NRTidalv2'  0.262977  0.142148   698.119175 -0.162536   
317   b'IMRPhenomPv2_NRTidalv2'  0.466045  0.145648   716.863844  0.088015   
422   b'IMRPhenomPv2_NRTidalv2'  0.315038  0.121040   586.604290  0.678848   
443   b'IMRPhenomPv2_NRTidalv2'  0.182732  0.086825   411.456267 -0.047396   
461   b'IMRPhenomPv2_NRTidalv2'  0.055646  0.038217   175.081647  0.068274   
545   b'IMRPhenomPv2_NRTidalv2'  0.259735  0.088182   418.265522  0.062869   
574   b'IMRPhenomPv2_NRTidalv2'  0.558054  0.052679   243.840155 -0.036032   
611   b'IMRPhenomPv2_NRTidalv2'  0.570572  0.159026   789.137985 -0.103791   
658   b'IMRPhenomPv2_NRTidalv2'  0.366650  0.099394   474.973689 -0.024779   
737   b'IMRPhenomPv2_NRTidalv2'  0.458404  0.101549   485.963795  0.370696   
805   b'IMRPhenomPv2_NRTidalv2'  0.222281  0.064298   300.050989 -0.111012   
824   b'IMRPhenomPv2_NRTidalv2'  0.234658  0.054478   252.490508 -0.064907   
836   b'IMRPhenomPv2_NRTidalv2'  0.590470  0.276049  1462.615539 -0.019214   
957   b'IMRPhenomPv2_NRTidalv2'  0.333680  0.422715  2399.849189 -0.566227   
1022  b'IMRPhenomPv2_NRTidalv2'  0.370024  0.090269   428.763881 -0.165559   
1031  b'IMRPhenomPv2_NRTidalv2'  0.245185  0.301095  1615.767822  0.466235   
1103  b'IMRPhenomPv2_NRTidalv2'  0.370349  0.345081  1891.840258  0.203789  

In [22]:
df_inj_both_flag_a_attrs.to_csv('files/attributes_inj_flagged_both_a.csv', index=None)

### All Catalog Events

In [4]:
df_O3 = pd.read_csv('files/O3_catalog.csv')

In [5]:
df_O3

category                  created em_events  em_type           far  \
0   Production  2020-08-26 19:12:58 UTC        []      NaN  1.049214e-09   
1   Production  2020-08-26 19:12:56 UTC        []      NaN  6.392341e-10   
2   Production  2020-08-26 19:12:54 UTC        []      NaN  2.030996e-18   
3   Production  2020-08-26 19:12:52 UTC        []      NaN  3.278744e-16   
4   Production  2020-08-26 19:12:50 UTC        []      NaN  5.977392e-13   
..         ...                      ...       ...      ...           ...   
69  Production  2021-10-22 17:39:42 UTC        []      NaN  1.013292e-07   
70  Production  2021-10-22 17:37:36 UTC        []      NaN  8.173553e-07   
71  Production  2021-10-22 17:37:18 UTC        []      NaN  5.697955e-12   
72  Production  2021-10-22 17:36:49 UTC        []      NaN  3.751973e-10   
73  Production  2021-10-22 17:36:39 UTC        []      NaN  1.452192e-08   

                                            gw_events             gw_id  \
0                         ['G6975', 'G6974', 'G6973']         GW190930A   
1                                  ['G6981', 'G6980']         GW190929A   
2                         ['G6953', 'G6952', 'G6951']         GW190924A   
3                ['G6979', 'G6978', 'G6977', 'G6976']         GW190915A   
4                                           ['G7009']         GW190910A   
..                                                ...               ...   
69           ['G73534', 'G73533', 'G73532', 'G73531']  GW191126C_o3bcat   
70                     ['G72907', 'G72906', 'G72905']  GW191113B_o3bcat   
71  ['G73398', 'G73397', 'G73396', 'G73395', 'G733...  GW191109A_o3bcat   
72           ['G73285', 'G73284', 'G73283', 'G73282']  GW191105C_o3bcat   
73                     ['G73939', 'G73938', 'G73937']  GW191103A_o3bcat   

         ifo                                           ifo_snrs  \
0      H1,L1                 {'H1': 6.2468519, 'L1': 7.5997348}   
1   H1,L1,V1  {'H1': 5.8978395, 'L1': 7.6792221, 'V1': 1.942...   
2   H1,L1,V1  {'H1': 5.9733548, 'L1': 11.240391, 'V1': 3.336...   
3   H1,L1,V1  {'H1': 9.5910568, 'L1': 8.5375166, 'V1': 2.449...   
4      L1,V1                 {'L1': 13.205681, 'V1': 2.3740058}   
..       ...                                                ...   
69     H1,L1                 {'H1': 5.8211088, 'L1': 6.2469616}   
70  H1,L1,V1   {'H1': 6.263817, 'L1': 6.384114, 'V1': 2.194304}   
71     H1,L1                  {'H1': 8.580859, 'L1': 12.552604}   
72     H1,L1                 {'H1': 5.8810124, 'L1': 7.7976589}   
73     H1,L1                  {'H1': 6.2123165, 'L1': 6.878675}   

                                               labels  ... space_coinc_far  \
0   ['GDB_YES', 'PE_YES', 'DETCHAR_YES', 'PRELIM',...  ...             NaN   
1   ['GDB_YES', 'PE_YES', 'DETCHAR_YES', 'NONE', '...  ...             NaN   
2   ['GDB_YES', 'PE_YES', 'DETCHAR_YES', 'PRELIM',...  ...             NaN   
3   ['GDB_YES', 'PE_YES', 'DETCHAR_YES', 'PRELIM',...  ...             NaN   
4   ['GDB_YES', 'PE_YES', 'DETCHAR_YES', 'PRELIM',...  ...             NaN   
..                                                ...  ...             ...   
69  ['GDB_YES', 'PE_NO', 'DETCHAR_YES', 'NONE', 'C...  ...             NaN   
70  ['GDB_YES', 'PE_NO', 'DETCHAR_NO', 'NONE', 'CH...  ...             NaN   
71  ['GDB_YES', 'PE_NO', 'DETCHAR_YES', 'NONE', 'C...  ...             NaN   
72  ['GDB_YES', 'PE_NO', 'DETCHAR_YES', 'NONE', 'C...  ...             NaN   
73  ['GDB_YES', 'PE_NO', 'DETCHAR_YES', 'NONE', 'C...  ...             NaN   

                 submitter  superevent_id           t_0         t_end  \
0      chad.hanna@LIGO.ORG       S190930a  1.253886e+09  1.253886e+09   
1      chad.hanna@LIGO.ORG       S190929a  1.253755e+09  1.253755e+09   
2      chad.hanna@LIGO.ORG       S190924a  1.253327e+09  1.253327e+09   
3      chad.hanna@LIGO.ORG       S190915a  1.252627e+09  1.252627e+09   
4      chad.hanna@LIGO.ORG       S190910a  1.252150e+09  1.252150e+09   
..          

In [6]:
gwids = ['_'.join([df_O3['gw_id'].iloc[i][:8],
                   from_gps(df_O3['t_0'].iloc[i]).strftime('%H%M%S')]) for i in range(len(df_O3))]

In [7]:
df_O3['name'] = gwids

In [8]:
offline_names_O3b = ['GW191103_012549', 'GW191113_071753', 'GW191126_115259', 'GW191127_050227',
 'GW191204_110529', 'GW191219_163120', 'GW191230_180458', 'GW200202_154313',
 'GW200208_222617', 'GW200209_085452', 'GW200210_092254', 'GW200216_220804',
 'GW200220_061928', 'GW200220_124850', 'GW200306_093714', 'GW200308_173609',
 'GW200322_091133']

In [9]:
offline_names_O3a = ['GW190413_052954','GW190413_134308','GW190424_180648','GW190514_065416',
                    'GW190527_092055','GW190620_030421', 'GW190708_232457', 'GW190719_215514',
                    'GW190731_140936', 'GW190803_022701', 'GW190909_114149', 'GW190910_112807',
                    'GW190929_012149']

### These names are taken from Table 1 and Table 4 of GWTC 3 and GWTC 2 paper (https://arxiv.org/pdf/2010.14527.pdf) respectively

In [10]:
df_O3b_offline = df_O3[df_O3['name'].isin(offline_names_O3b)]
df_O3b_online = df_O3[(~df_O3['name'].isin(offline_names_O3b)) & (df_O3['t_0']>=to_gps('2019-11-01'))]

In [11]:
df_O3a_offline = df_O3[df_O3['name'].isin(offline_names_O3a)]
df_O3a_online = df_O3[(~df_O3['name'].isin(offline_names_O3a)) & (df_O3['t_0']<to_gps('2019-11-01'))]

In [12]:
len(df_O3a_offline), len(df_O3a_online)

(13, 26)

In [13]:
len(df_O3b_offline), len(df_O3b_online)

(17, 18)

In [17]:
df_O3a_online.to_csv('files/O3a_online.csv', index=None)
df_O3b_online.to_csv('files/O3b_online.csv', index=None)
df_O3a_offline.to_csv('files/O3a_offline.csv', index=None)
df_O3b_offline.to_csv('files/O3b_offline.csv', index=None)

### Retractions

In [15]:
df_retr = pd.read_csv('files/retractions_O3.csv')

In [16]:
df_retr

superevent_id                  created submitter preferred_event  \
0       S200308e  2020-03-08 01:20:11 UTC  emfollow         G367118   
1      S200303ba  2020-03-03 12:16:14 UTC  emfollow         G366412   
2      S200116ah  2020-01-16 11:57:11 UTC  emfollow         G360499   
3       S200108v  2020-01-08 10:01:07 UTC  emfollow         G359656   
4      S200106av  2020-01-06 18:36:27 UTC  emfollow         G359542   
5      S200106au  2020-01-06 18:35:21 UTC  emfollow         G359544   
6      S191225aq  2019-12-25 21:57:52 UTC  emfollow         G358412   
7      S191220af  2019-12-20 12:24:45 UTC  emfollow         G357916   
8      S191213ai  2019-12-13 15:59:37 UTC  emfollow         G357245   
9       S191212q  2019-12-12 08:28:21 UTC  emfollow         G357129   
10     S191124be  2019-11-24 10:00:11 UTC  emfollow         G355462   
11     S191120at  2019-11-20 20:09:03 UTC  emfollow         G355100   
12     S191120aj  2019-11-20 16:23:51 UTC  emfollow         G355085   
13      S191117j  2019-11-17 06:08:46 UTC  emfollow         G354833   
14     S191110af  2019-11-10 23:10:59 UTC  emfollow         G354360   
15      S191110x  2019-11-10 18:09:05 UTC  emfollow         G354345   
16      S190928c  2019-09-28 02:14:18 UTC  emfollow         G351785   
17      S190829u  2019-08-29 21:06:19 UTC  emfollow         G348652   
18      S190822c  2019-08-22 01:30:23 UTC  emfollow         G347846   
19      S190816i  2019-08-16 13:05:12 UTC  emfollow         G347380   
20     S190808ae  2019-08-08 22:21:45 UTC  emfollow         G346647   
21      S190524q  2019-05-24 04:52:30 UTC  emfollow         G333983   
22     S190518bb  2019-05-18 19:19:39 UTC  emfollow         G333313   
23     S190405ar  2019-04-05 16:01:56 UTC  emfollow         G328997   

         t_start           t_0         t_end  \
0   1.267666e+09  1.267666e+09  1.267666e+09   
1   1.267273e+09  1.267273e+09  1.267273e+09   
2   1.263211e+09  1.263211e+09  1.263211e+09   
3   1.262513e+09  1.262513e+09  1.262513e+09   
4   1.262371e+09  1.262371e+09  1.262371e+09   
5   1.262371e+09  1.262371e+09  1.262371e+09   
6   1.261346e+09  1.261346e+09  1.261346e+09   
7   1.260880e+09  1.260880e+09  1.260880e+09   
8   1.260288e+09  1.260288e+09  1.260288e+09   
9   1.260174e+09  1.260174e+09  1.260174e+09   
10  1.258625e+09  1.258625e+09  1.258625e+09   
11  1.258316e+09  1.258316e+09  1.258316e+09   
12  1.258302e+09  1.258302e+09  1.258302e+09   
13  1.258006e+09  1.258006e+09  1.258006e+09   
14  1.257462e+09  1.257462e+09  1.257462e+09   
15  1.257445e+09  1.257445e+09  1.257445e+09   
16  1.253672e+09  1.253672e+09  1.253672e+09   
17  1.251148e+09  1.251148e+09  1.251148e+09   
18  1.250473e+09  1.250473e+09  1.250473e+09   
19  1.249996e+09  1.249996e+09  1.249996e+09   
20  1.249338e+09  1.249338e+09  1.249338e+09   
21  1.242709e+09  1.242709e+09  1.242709e+09   
22  1.242242e+09  1.242242e+09  1.242242e+09   
23  1.238515e+09  1.238515e+09  1.238515e+09   

                                       gw_events           far  \
0                                    ['G367118']  3.618608e-09   
1              ['G366412', 'G366411', 'G366410']  1.315616e-08   
2                                    ['G360499']  2.028966e-12   
3                         ['G359657', 'G359656']  2.668899e-13   
4                         ['G359542', 'G359541']  3.127682e-08   
5              ['G359544', 'G359543', 'G359540']  3.660501e-09   
6              ['G377807', 'G358412', 'G358411']  1.267018e-08   
7                         ['G357916', 'G357915']  3.963175e-10   
8                                    ['G357245']  2.003527e-08   
9                                    ['G357129']  3.368703e-08   
10                                   ['G355462']  1.678100e-09   
11             ['G355101', 'G355100', 'G355099']  6.109056e-09   
12                        ['G355085', 'G355084']  2.860121e-08   
13                                   ['G354833']  1.114482e-18   
14                                  